lib/parser/builders/default.rb

```diff
@@ -2246,7 +2246,7 @@ def static_regexp(parts, options)
         source
       end
 
-      Regexp.new(source, (Regexp::EXTENDED if options.children.include?(:x)))
+      Regexp.new(source.to_s, (Regexp::EXTENDED if options.children.include?(:x)))
```

Comment:
Please correct if I'm wrong, but I think `source` is always a `String` on this line, and so calling `.to_s` should be a noop.

Reply:
You're correct -- at least when I put a breakpoint before Line 2249 -- I can see it's a string but for some reason it doesn't cast unless you message `#to_s`.